core/src/main/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunction.java

```diff
@@ -500,26 +500,22 @@ private static DefaultFunctionResolver truncate() {
         FunctionDSL.impl(
             FunctionDSL.nullMissingHandling(
                 (x, y) -> new ExprLongValue(
-                    new BigDecimal(x.integerValue()).setScale(y.integerValue(),
-                        RoundingMode.DOWN).longValue())),
+                        MathUtil.truncateNumber(x.integerValue(), y.integerValue()).longValue())),
```

Comment:
this is going to convert `x.integerValue()` into a `double` since `truncateNumber` only accepts `double`.  We should have a `truncateNumber` function for `int`, `long`, `float`, etc. I'm not sure this saves much effort/code?